docs/basic/flattening.md

```diff
@@ -0,0 +1,134 @@
+would mean you would get the wrong heights for your data.
+
+## Masking
+
+Now consider that all the processing we have done has assumed that every pixel of the image is background. We assumed
+that there were no objects on the surface, messing up our fitting, and row alignment. If there was a large amount of
+DNA on one side of the image, then the slope will be affected by it, and so flatten the image poorly.
+
+Because of this, once we have done our initial flattening, we detect our objects on the surface, and then flatten the
+image again! But this time, ignoring the data on the surface, and only considering the background.
+
+How do we do that?
+Well first, we need to find the data on the surface. We do this by thresholding.
+The threshold is set by the config file (have a look!). Any pixels that are below the threshold, are considered
+background (sample surface). Any pixels that are above the threshold are considered to be data (useful sample objects).
+This binary classification allows us to make a binary mask of where is foreground data, and where is background.
+
+For more information on thresholding and how to set it, see the [thresholding](thresholding.md) page.
+
+Here is the binary mask for minicircle.spm:
+
+![processed image so far](../_static/images/flattening/flattening_tilt_removal.png)
+
+![binary mask](../_static/images/flattening/flattening_binary_mask.png)
+
+So you can see how all the interesting high up parts are now masked in white, and the background is in black.
+
+TopoStats can use this to only use the black regions in its calculations for slope removal, row alignment etc.
+
+So we re-do all the previous processing, but with this new useful binary mask to guide us.
+
+## Secondary flattening
+
+After re-processing the data using the mask to tell us where the background is, we get a better, more accurately
+flattened image.
+
+From here, we can go on to do things like finding our objects of interest (grains) and get stats about them.
+
+![secondary flattening](../_static/images/flattening/flattening_final_flattened_image.png)
+
+## Gaussian filter
+
+Finally, we apply a Gaussian filter to the image to remove jitteriness and noise. This allows you to get smoother data
+but will start to blur out important features if you apply it too strongly. The default strength is a sigma of 1.0, but
+you can adjust this in the config file under `filter/gaussian_size`.
+
+Here are some examples of different gaussian sizes:
+
+### Strength: 0.01
+
+![gaussian_filter_0](../_static/images/flattening/flattening_gaussian_filtered_0.png)
+
+### Strength: 1.0
+
+![gaussian_filter_01](../_static/images/flattening/flattening_gaussian_filtered_01.png)
+
+### Strength: 2.0
+
+![gaussian_filter_02](../_static/images/flattening/flattening_gaussian_filtered_02.png)
+
+### Strength: 8.0
+
+![gaussian_filter_08](../_static/images/flattening/flattening_gaussian_filtered_08.png)
```

Comment:
Would it be easier to see if these were side-by-side like in a figure done in illustrator?